autopilot/autopilot.go

```diff
@@ -176,12 +192,16 @@ func (ap *Autopilot) Run() error {
 }
 
 func (ap *Autopilot) Stop() error {
-	if ap.ticker != nil {
+	ap.mu.Lock()
+	defer ap.mu.Unlock()
```

Comment:
Holding a lock while waiting is a recipe for a deadlock. In case anything needs to quickly grab the lock after receiving the stop signal.

Reply:
I think it should wrap the `wait` because it's a mutex that controls autopilot start/stop willy-nilly. I'll just rename it